src/EVM/Types.hs

```diff
@@ -680,6 +683,14 @@ data FrameState = FrameState
   }
   deriving (Show, Generic)
 
+data Memory s
+  = ConcreteMemory !(STVector s Word8)
+  | SymbolicMemory !(Expr Buf)
+
+instance Show (Memory s) where
+  show (ConcreteMemory _) = "<can't show mutable memory>"
```

Comment:
why not?

Reply:
because we need to be in ST or IO monad to do that